modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -222,6 +267,29 @@ private void onPlacementDriverMessageReceived(NetworkMessage msg0, String sender
         }
     }
 
+    /**
+     * Sends stop lease prolongation message to all participants of placement driver group.
+     *
+     * @param groupId Replication group id.
+     * @param redirectNodeId Node consistent id to redirect.
+     */
+    private void declineLeaseProlongation(ReplicationGroupId groupId, String redirectNodeId) {
+        LOG.info("The replica does not meet the requirements for the leaseholder [groupId={}, redirectNodeId={}]", groupId, redirectNodeId);
+
+        msNodes.thenAccept(nodeIds -> {
```

Comment:
also, the leaderFuture on replica must be reset in order to hold on next LeaseGrantedMessage and wait for the next leader election